### PR TITLE
feat(explore): UI changes in dataset panel on Explore page

### PR DESCRIPTION
**Pull request by kgabryje**
### SUMMARY
This PR implements several UI changes in Explore's dataset panel.
List of changes:
- change color of background to white
- separate each column/metric label with vertical margins
- add dots icon to column/metric labels to indicate they're draggable
- move certification icon to the right side of the label
- Change font size of column/metric section labels to 12 and remove separation line

### BEFORE/AFTER SCREENSHOTS OR ANIMATED GIF
image

### TESTING INSTRUCTIONS
Verify that everything works like before

### ADDITIONAL INFORMATION

- [ ] Has associated issue:
- [ ] Required feature flags:
- [ ] Changes UI
- [ ] Includes DB Migration (follow approval process in [SIP-59](https://github.com/apache/superset/issues/13351))
  - [ ] Migration is atomic, supports rollback & is backwards-compatible
  - [ ] Confirm DB migration upgrade and downgrade tested
  - [ ] Runtime estimates and downtime expectations provided
- [ ] Introduces new feature or API
- [ ] Removes existing feature or API

CC @kasiazjc

### CODE REVIEW

**Comment by michael-s-molina** [comment]
I loved the changes! Really improved the user experience. cc @kasiazjc 

I have one suggestion that may be tackled in a separate PR if you wish. Can we make the draggable image respect the border-radius and width of the original component? If we do that, when the user is dropping, it will fit exactly in the droppable area 🤩 

https://user-images.githubusercontent.com/70410625/160459019-b4671a3c-56d7-449c-8da2-e218ec7ebd86.mov

File: superset-frontend/src/components/Icons/index.tsx [hunk 1]
```diff
@@ -65,6 +65,7 @@ const IconFileNames = [
   'dataset_virtual',
   'download',
   'drag',
+  'drag_icon',
```
Comment (michael-s-molina) [inline superset-frontend/src/components/Icons/index.tsx:68]: I'm not sure `drag_icon` is the best name, all declarations here are icons. What's the difference between the drag icons? Maybe we can come up with a better name.

Comment (kgabryje) [reply superset-frontend/src/components/Icons/index.tsx:68]: Nahh, I just forgot to delete that (added for test purposes). Good eye as always 😄

**Comment by kgabryje** [comment]
> Can we make the draggable image respect the border-radius and width of the original component

Good point, done! 
image

File: superset-frontend/src/explore/components/DatasourcePanel/index.tsx [hunk 2]
```diff
@@ -97,26 +101,55 @@ const DatasourceContainer = styled.div`
...
 const LabelWrapper = styled.div`
-  overflow: hidden;
-  text-overflow: ellipsis;
+  ${({ theme }) => css`
+    overflow: hidden;
+    text-overflow: ellipsis;
+    font-size: ${theme.typography.sizes.s}px;
+    background-color: ${theme.colors.grayscale.light4};
+    margin: ${theme.gridUnit * 2}px 0;
+    border-radius: 4px;
```
Comment (michael-s-molina) [inline superset-frontend/src/explore/components/DatasourcePanel/index.tsx:110]:
```suggestion
    border-radius: ${theme.gridUnit}px;
```

File: superset-frontend/src/explore/components/DatasourcePanel/index.tsx [hunk 3]
```diff
@@ -273,13 +302,13 @@ export default function DataSourcePanel({
         />
         <div className="field-selections">
           <Collapse
-            bordered
+            // bordered
```
Comment (michael-s-molina) [inline superset-frontend/src/explore/components/DatasourcePanel/index.tsx:305]:
```suggestion
```

File: superset-frontend/src/explore/components/DatasourcePanel/index.tsx (hunk 2 above)
Comment (kgabryje) [reply superset-frontend/src/explore/components/DatasourcePanel/index.tsx:110]: I don't think that is related to gridUnit. Supposedly, if we created a theme where distances between components are bigger, that doesn't mean we want the corners to be more rounded

File: superset-frontend/src/dashboard/components/dnd/DragHandle.tsx [hunk 4]
```diff
@@ -17,33 +17,35 @@
...
 interface DragHandleProps {
   position: 'left' | 'top';
-  innerRef: LegacyRef<HTMLDivElement> | undefined;
-  dotCount: number;
+  innerRef?: LegacyRef<HTMLDivElement> | undefined;
 }
 
+const DragHandleContainer = styled.div<{ position: 'left' | 'top' }>`
+  ${({ theme, position }) => css`
+    margin-bottom: ${theme.gridUnit}px !important;
```
Comment (michael-s-molina) [inline superset-frontend/src/dashboard/components/dnd/DragHandle.tsx:30]:
```suggestion
    height: 21px;
```

Comment (kgabryje) [reply superset-frontend/src/dashboard/components/dnd/DragHandle.tsx]: 👍 
Looks good after changing to height: 20, thanks for the tip
image

**Review by michael-s-molina — APPROVED** [review]
LGTM. Thanks for the amazing changes @kgabryje! It looks so much better!

**Pull request merged**